### DISCUSSION TITLE
API Button

### DISCUSSION THREAD

**Discussion by craigwitter** [HASwitchPlate/openHASP#504] [Q&A] [answered]
Hello all,
  I'm wondering if there is a way to make a button that can trigger a webservice call directly, vs having to go through HomeAssistant?   My thought it be able to call a shelly that's in the ceiling directly via a button. I was reading about the offline options that would normally communicate with just the relays on the actual switch. My assumption would be that the wifi would be available and this would work regardless of whether HomeAssistant was available at that moment or not.  

Thanks in advance,

Craig

**Comment by fvanroie** [comment, marked as answer]
openHASP isn't really conceived with direct control in mind. It behaves as an endpoint, like a shelly, and not like a controller...

That being said, the only way to make a HTTP request is to set the `src` of an `image` object as it should try to download the image from the webserver. You could *in theory* make an invisible `img` object and set the `src` attribute on `up` event (or `down`) in firmware version 0.7.0-rc4/5.

Another *workaround* is to customize the firmware directly to your needs by compiling your own.

**Comment by adorobis** [comment]
Cool workaround :)
@craigwitter have you managed to implement it in the device directly? I can't figure that out from the docs so far.
